### ISSUE TITLE
flySystem permission issues

### ISSUE THREAD

**Issue by matdave** [modxcms/revolution#15725]
### Summary
The flySystem is clearing out file permissions if the file's original permissions aren't the "expected" permissions. 

### Step to reproduce
Here is how I am able to recreate it on my system. 

```
~/path/to/file/$ chmod 666 file.php
``` 

Make a change to that file in MODX. 

```
~/path/to/file/$ ls file.php
----------    1 c0XXX    c0XXX         8053 May 24 10:44 file.php
```
But with "correct" permissions: 

```
~/path/to/file/$ chmod 644 file.php
``` 
Make a change to that file in MODX.

```
~/path/to/file/$ ls file.php
-rw-r--r--    1 c0XXX    c0XXX         8053 May 24 10:44 file.php
```

### Observed behavior
It seems the flySystem is getting confused and just dropping the permissions entirely. 

### Expected behavior
It should either set the file to the "correct" permissions or leave them alone. 

### Environment
MODX 3.0.0-apha3 on MODX Cloud

**Comment by opengeek** [comment]
Unfortunately, I believe this would need to be dealt with at the FlySystem project, and I don't believe v1 which is in use here is even being maintained any longer.

**Comment by Ibochkarev** [comment]
@matdave Is it not fixed in #15757 ?

**Comment by matdave** [comment]
@Ibochkarev confirmed fixed
